streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -300,7 +300,7 @@ private long processNext(final List<ConsumerRecord<byte[], byte[]>> records,
             }
             lastRestoredOffset = offset;
             numberRestored++;
-            if (record.key() != null) {
+            if (record.key() != null && record.value() != null) {
```

Comment:
This does not seem to be correct -- if `value -- null` we still need to include the record as delete in the restore.